examples/llm/main.py

```diff
@@ -64,6 +69,23 @@ def main(cfg):
             "Reverting to `cfg.model.init_device='cpu'`.")
         cfg.model.init_device = 'cpu'
 
+    te_config = cfg.get('te_config', None)
+    te_config = om.to_container(te_config, resolve=True) if te_config else None
+    if not te_installed and te_config:
+        warnings.warn(
+                "Transformer Engine is not installed but te_config is given! "
+                "Please install transformer engine with "
+                "pip install --upgrade git+https://github.com/NVIDIA/TransformerEngine.git@stable")
+    if not te_installed and not te_config['linear_only']:
+        print('use attn_impl == Triton to take care of key_padding_mask with TransformerEngine...')
```

Comment:
log instead of print. Message should also be more explicit: "Overriding attn_impl to use Triton to be compatible with Transformer Engine."